vm_method.c

```diff
@@ -1378,7 +1386,7 @@ rb_export_method(VALUE klass, ID name, rb_method_visibility_t visi)
     VALUE defined_class;
     VALUE origin_class = RCLASS_ORIGIN(klass);
 
-    me = search_method(origin_class, name, &defined_class);
+    me = search_method0(origin_class, name, &defined_class, VM_METHOD_TYPE_REFINED);
```

Comment:
if skip type is only REFINED, how about to make `skip_refined` boolean flag instead of `skip_type`?

Reply:
The approach of taking a type is a bit more flexible, in case we want to skip other method types in other cases.  However, I can make that change if you prefer.  This isn't public API, so we could always change it later if needed.

Reply:
At least, 0 is a possible valid value as `rb_method_type_t`.

Reply:
I originally tried to use -1 instead of 0, but it complained about that :).  From the looks of it, using `END_OF_VM_METHOD_TYPE_PLACEHOLDER` may not be portable, and `VM_METHOD_TYPE_REFINED+1` will just break later if we expand the values in `rb_method_type_t`.  So I'll go with @ko1's suggestion.